docs/v1/api.md

```diff
@@ -0,0 +1,333 @@
+the aforementioned Kubernetes resources. The service catalog controller watches
+these resources and takes action on them.
+
+Like ingress controllers, the operator is free to choose any service controller
+they prefer. All implementations, however, must satisfy the following
+requirements:
+
+- Watch the event stream for new or deleted `ServiceInstance`s and
+  `ServiceInstanceClaim`s
+- On `ServiceInstance` creation, the controller must do a provision
+- On `ServiceInstance` deletion, the controller must do a deprovision. The
+  controller should not delete any existing `ServiceInstanceClaim`s or
+  `ServiceInstanceBinding`s that reference the deleted `ServiceInstance`
+- On `ServiceInstanceClaim` creation:
+  - If the claim references an existing `ServiceInstance`, the controller must
+    do a bind
+  - If the claim references an existing `ServiceClass`, the controller must do
+    a provision, then bind. The controller should not create a
+    publicly-accessible `ServiceInstance`. This ensures that the
+    provisioned/bound resource will be single-tenant and dedicated
+- On `ServiceInstanceClaim` deletion:
+  - If the claim referenced an existing `ServiceInstance`, the controller must
+    do an unbind
+  - If the claim referenced an existing `ServiceClass`, the controller must do
+    an unbind, then a deprovision
+- Any successful provision operation must change the `status` field of the
+  applicable `ServiceInstanceClaim` to `provisioned`
+- Any successful bind operation must:
+  - Change the `ServiceInstanceClaim`'s `status` field to `bound`
+  - Create the appropriate resources (`Secret`s, `ConfigMap`s) to hold
+    credential and other resource-specific bind information
+  - Create a new `ServiceInstanceBinding`:
+    - With the same name as the claim’s `targetBinding` field
+    - In the same namespace as the claim itself
+    - With fields that point to all the previously created resources
+- Any successful unbind operation must:
+  - Change the claim’s `status` field to `unbound`
+  - Delete the `ServiceInstanceBinding` that was created in the bind operation
+  - Delete all resources that the aforementioned `ServiceInstanceBinding`
+    points to
+- Any failed operation must change the status field of the
+  `ServiceInstanceClaim` to `failed`, and should write the reason of the
+  failure into the `statusDescription` field
+
+The below diagram shows all of the interactions that all service catalog
+controllers are expected to have.
+
+![Complete System](./img/entire-flow.png)
+
+# Visibility
+
+The visibility of each resource and system described above is important to the
+functionality and ease-of-use in this system.
+
+The aforementioned Kubernetes resources share many of the same visibility rules
+as those in the persistent volumes and ingress systems.
+
+- `ServiceClasse`s are cluster-global (just as `StorageClass`es are)
+- `ServiceInstance`s are cluster-global (just as `PersistentVolume`s are)
```

Comment:
Are ServiceInstances are more like Kubernetes Services (created by the user, namespaced) or like automatically provisioned PersistentVolumes (created on demand by the cluster in response to user need, cluster-scoped)? In both cases it would be possible to access the ServiceInstance across namespaces. Our DNS hierarchy supports that, for instance.